Tests/mock_server.py

```diff
@@ -362,101 +371,159 @@ def start(self, playbook_id, path=None, record=False, thread_index=0, prints_man
-            actions = '-s {} '.format(remote_script_path)
-            actions += '--set script_mode=playback '
-            actions += '--set keys_filepath={} --server-replay-kill-extra --server-replay'.format(
-                repo_problem_keys_filepath
-            )
+            silence_output(self.ami.call,
+                           ['mv', repo_problem_keys_path, current_problem_keys_path],
+                           stdout='null')
+        return self._write_mitmdump_rc_file_to_host(file_content, prints_manager, thread_index)
+
+    def _write_mitmdump_rc_file_to_host(self,
+                                        file_content: str,
+                                        prints_manager,
+                                        thread_index: int) -> bool:
+        """
+        Does all needed preparation for starting the proxy service which include:
+        - Creating the mitmdump_rc file that includes the script mode, keys file path, mock file path and log file path
+          for the mitmdump service and puts it in '/home/ec2-user/mitmdump_rc' in the remote machine.
 
-        log_file = os.path.join(path, get_log_file_path(playbook_id, record))
-        # Handle proxy log output
-        debug_opt = " >{} 2>&1".format(log_file) if not self.debug else ''
+        Args:
+            file_content: The content of the mitmdump_rc file that includes the script mode, keys file path,
+            mock file path and log file path
+            prints_manager: A prints manager instance
+            thread_index: The index of the current thread
 
-        # all mitmproxy/mitmdump commands should have the 'source .bash_profile && ' prefix to ensure the PATH
-        # is correct when executing the command over ssh
-        # Configure proxy server
-        command = "source .bash_profile && mitmdump --ssl-insecure --verbose --listen-port {} {} {}{}".format(
-            self.PROXY_PORT, actions, os.path.join(path, get_mock_file_path(playbook_id)), debug_opt
-        )
-        command = command.split()
+        Returns:
+            True if file was successfully copied to the server, else False
+        """
+        with open('mitmdump_rc', 'w') as mitmdump_rc:
+            mitmdump_rc.write(file_content)
+        try:
+            self.ami.copy_file('mitmdump_rc', dst=AMIConnection.REMOTE_HOME)
+            return True
+        except CalledProcessError as err:
+            prints_manager.add_print_job(f'Could not copy arg file for mitmdump service to server {self.ami.public_ip},'
+                                         f' Error: {err}',
+                                         print_error,
+                                         thread_index)
+        return False
+
+    @timeout(5, exception_message='Mitmdump failed to start after 5 seconds', use_signals=False)
+    def wait_until_proxy_is_listening(self):
+        """
+        Checks if the mitmdump service is listening, and raises an exception if 5 minutes pass without positive answer
```

Comment:
Your description here says 5 minutes, but the `exception_message` passed to the decorator says 5 seconds. Please fix whichever of them is incorrect.